app/controllers/application_controller.rb

```diff
@@ -1,10 +1,160 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    set :session_secret, "password_security"
+    enable :sessions
+  end
+
+  get '/' do
+    @username = username
+    erb :index
+  end
+
+  #----------------SESSIONS--------------
+
+  get '/login' do
+    if logged_in?
+      redirect to "/tweets"
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+    if user && user.authenticate(params[:password])
+      session[:id] = user.id
+      redirect to "/tweets"
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/logout' do
+    session.clear
+    redirect "/login"
+  end
+
+  #-----------------CREATE-----------------
+
+  get '/signup' do ### CREATE USER
+    if logged_in?
+      redirect to "/tweets"
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params["username"].empty? || params["email"].empty? || params["password"].empty?
+      redirect to "/signup"
+    else
+      @user = User.create(username: params[:username], email: params[:email], password: params[:password])
+      session[:id] = @user.id
+      session[:username] = @user.username
+      redirect to "/tweets"
+    end
+  end
+
+  get '/tweets/new' do ### CREATE TWEET
+    if logged_in?
+      erb :'tweets/create_tweet'
+    else
+      redirect to "/login"
+    end
+  end
+
+  post '/tweets' do
+    if !params["content"].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.